web3/pm.py

```diff
@@ -245,11 +246,12 @@ def _get_package_name(self, package_id: bytes) -> str:
     @to_tuple
     def _get_all_package_ids(self) -> Iterable[Tuple[bytes]]:
         num_packages = self._num_package_ids()
+        pointer = 0
+        package_ids = []
+        while pointer < num_packages:
+            new_ids, pointer = self.registry.functions.getAllPackageIds(pointer, (pointer + BATCH_SIZE)).call()
+            package_ids.append(new_ids)
```

Comment:
@pipermerriam Not sure how you feel about this - feels a bit mutate-y to me, but also the easiest way to implement it. The `BATCH_SIZE` is also arbitrary, but I tested it locally with values of `1`, `2`, `4`, `10`, and `100` and it works fine. I didn't update the tests in this case b/c releasing 100+ packages just to test this case seems a bit overkill.

Reply:
This should be able to be converted to a non-mutative solution reasonably easy.  IIUC this correctly you need to reverse each of the individual chunks and return in order which I think would be equivalent to just replacing this statement with `yield from reversed(new_ids)`